doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -441,6 +441,109 @@ Here are examples:
    end
    where "n + m" := (plus n m).
 
+Enabling and disabling notations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. cmd:: {| Enable | Disable } Notation @toggle_notation
+   :name: Enable Notation; Disable Notation
+
+   .. insertprodn toggle_notation toggle_notation_flag
+
+   .. prodn::
+      toggle_notation ::= {| @string | @qualid } := @one_term {? ( {+, @toggle_notation_flag } ) } {| : no scope | : @ident }
+      | {| @string | @qualid } := @one_term {? ( {+, @toggle_notation_flag } ) }
+      | {| @string | @qualid } {? ( {+, @toggle_notation_flag } ) } {| : no scope | : @ident }
+      | {| @string | @qualid } {? ( {+, @toggle_notation_flag } ) }
+      | := @one_term {? ( {+, @toggle_notation_flag } ) } {| : no scope | : @ident }
+      | := @one_term {? ( {+, @toggle_notation_flag } ) }
+      | {? ( {+, @toggle_notation_flag } ) } {| : no scope | : @ident }
+      toggle_notation_flag ::= all
+      | only parsing
+      | only printing
+      | in custom @ident
+      | in constr
+
+   Enables or disables notations previously defined with
+   :cmd:`Notation` or :cmd:`Notation (abbreviation)`.
+   This command has no effect on parser modifications made with
+   :cmd:`Reserved Notation`.
+   When multiple
+   informations are provided, the notations enabled or
+   disabled are the ones meeting the conjunction of all constraints.
+
+   This command supports the :attr:`local` and :attr:`global`
+   attributes.
+
+   :n:`@string`
+      Notations to enable or disable. :n:`@string` can be a single
+      token in the notation such as "`->`" or a pattern that matches
+      the notation. See :ref:`locating-notations`.
+
+   :n:`@qualid`
+      :ref:`Abbreviation <Abbreviations>` to enable or disable.
+
+   :n:`@one_term`
+      Enable or disable notations whose interpretation contains :token:`one_term`.
+
+   :n:`all`
+      Enable or disable all notations meeting the given constraints,
+      even if multiple ones. When not provided, there must be a single
+      notation meeting the constraints.
+
+   :n:`only parsing`
+      The notation is enabled or disabled only for parsing.
+
+   :n:`only printing`
+      The notation is enabled or disabled only for printing.
+
+   :n:`in custom @ident`
+      Enable or disable notations in the given :ref:`custom entry
+      <custom-entries>`.
+
+   :n:`in constr`
+      Enable or disable notations in ``constr``. See
+      :ref:`custom entries <custom-entries>`.
+
+   :n:`{? {| : no scope | : @ident } }`
+      If given, only notations in scope :token:`ident` are affected (or
```

Comment:
```suggestion
   :n:`{| : no scope | : @ident }`
      Only notations in scope :token:`ident` are affected (or
```

If `ident` is the more common case, I'd put that before `no scope` everywhere (mlgs, doc).  Your explanation here covers `ident` first.  And shouldn't `ident` be something like `scope` (scope_name, scope_key)?

Reply:
> If `ident` is the more common case, I'd put that before `no scope` everywhere (mlgs, doc). Your explanation here covers `ident` first.

Right, done

> And shouldn't `ident` be something like `scope` (scope_name, scope_key)?

It should indeed be `scope_name`, see above comment https://github.com/coq/coq/pull/12324#discussion_r1011379013